### PR TITLE
Test rating option schema

### PR DESCRIPTION
**Pull request by suhasdombale123**
## Why?
Added Test cases for Rating Option Schema 

## This change addresses the need by:

Added test cases in file schema/rating/rating_option_test.exs
added separate describe funtion for both funtions create_changset/2 and update_changeset/2 

[delivers #163248658]

## Checklist

- [x] I have read [CONTRIBUTING.md][contributing].
- [x] My code follows the [style guidelines][contributing] of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [x] I have updated the documentation wherever necessary.
- [x] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

### CODE REVIEW

File: apps/snitch_core/test/data/schema/rating/rating_option_test.exs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    position: 1,
+    rating_id: 105
+  }
+  describe "create_changeset/2 " do
+    test "succeeds" do
+      %{valid?: validity} = RatingOption.create_changeset(%RatingOption{}, @valid_attrs)
+      assert validity
+    end
+
+    test "fails? with missing code" do
```
Comment (arjun289) [inline apps/snitch_core/test/data/schema/rating/rating_option_test.exs]: All the `required` related params can be tested in just one test. No need to test failure for every single `key` in schema. 

Just pass empty params object to `create_changeset(%RatingOption{}, %{})` and you will get all the required related errors.

File: apps/snitch_core/test/data/schema/rating/rating_option_test.exs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  describe "update_changeset/2 " do
+    test "succeeds" do
+      param = Map.put(@valid_attrs, :position, 2)
+      %{valid?: validity} = RatingOption.update_changeset(%RatingOption{}, param)
+      assert validity
+    end
+
+    test "fails? without position" do
+      param = Map.delete(@valid_attrs, :position)
```
Comment (arjun289) [inline apps/snitch_core/test/data/schema/rating/rating_option_test.exs]: Not tested in the right manner, first of all, insert data and then try to update it and check, all the changesets are receiving empty structs.

File: apps/snitch_core/test/data/schema/rating/rating_option_test.exs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    end
+
+    test "fails? with all empty" do
+      %{valid?: validity} = RatingOption.create_changeset(%RatingOption{}, %{})
+      refute validity
+    end
+  end
+
+  describe "update_changeset/2 " do
+    test "succeeds" do
```
Comment (arjun289) [inline apps/snitch_core/test/data/schema/rating/rating_option_test.exs]: First, insert a record and then try to use `update_changeset` on it.

**Review by arjun289 — DISMISSED** [review]

File: apps/snitch_core/test/data/schema/rating/rating_option_test.exs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  describe "create_changeset/2 " do
+    test "succeeds" do
+      %{valid?: validity} = RatingOption.create_changeset(%RatingOption{}, @valid_attrs)
+      assert validity
+    end
+
+    test "fails? with all empty" do
+      c = %{valid?: validity} = RatingOption.create_changeset(%RatingOption{}, %{})
+      refute validity
+      require IEx
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/schema/rating/rating_option_test.exs]: Remove `require IEx`

File: apps/snitch_core/test/data/schema/rating/rating_option_test.exs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      rating_opt = insert(:rating_option)
+      c = RatingOption.update_changeset(rating_opt, %{position: nil, code: nil})
+      refute c.valid?
+      assert %{position: ["can't be blank"], code: ["can't be blank"]} == errors_on(c)
+    end
+
+    test "fails? with invalid parameters" do
+      rating_opt = insert(:rating_option)
+      c = RatingOption.update_changeset(rating_opt, %{value: 2})
+      assert c.changes == %{}
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/schema/rating/rating_option_test.exs]: If we are testing failure, then we should assert the errors

File: apps/snitch_core/test/data/schema/rating/rating_option_test.exs [hunk 6]
```diff
@@ -0,0 +1,48 @@
+defmodule Snitch.Data.Schema.RatingOptionTest do
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+  import Snitch.Factory
+  alias Snitch.Core.Tools.MultiTenancy.Repo
```
Comment (SagarKarwande) [inline apps/snitch_core/test/data/schema/rating/rating_option_test.exs]: No alias required as `Snitch.DataCase` already has aliased

**Review by SagarKarwande — DISMISSED** [review]

File: apps/snitch_core/test/data/schema/rating/rating_option_test.exs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  use Snitch.DataCase
+  import Snitch.Factory
+  alias Snitch.Data.Schema.RatingOption
+
+  setup :rating_options
+
+  @tag rating_option_count: 1
+  describe "create_changeset/2" do
+    test "returns a valid changeset", %{rating_options: [rating_option]} do
+      rating_option = Map.from_struct(rating_option)
```
Comment (arjun289) [inline apps/snitch_core/test/data/schema/rating/rating_option_test.exs]: This is not the right approach. Inserting the `rating_options` and then using the same. Please do it with params.

**Review by arjun289 — COMMENTED** [review]
Add, `foreign_key` constraint for `rating` in `rating_option_schema` and test it.

**Review by ashish173 — APPROVED** [review]

**Pull request merged**